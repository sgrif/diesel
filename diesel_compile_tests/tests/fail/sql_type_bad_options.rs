extern crate diesel;

use diesel::sql_types::SqlType;

#[derive(SqlType)]
#[diesel(postgres_type)]
struct Type1;

#[derive(SqlType)]
#[diesel(postgres_type(type_name = "foo", oid = 2, array_oid = 3))]
struct Type2;

#[derive(SqlType)]
#[diesel(postgres_type(oid = 2))]
struct Type3;

#[derive(SqlType)]
#[diesel(postgres_type(oid = 1, array_oid = "1"))]
struct Type4;

#[derive(SqlType)]
#[diesel(postgres_type(oid = 1, ary_oid = "1"))]
struct Type5;

#[derive(SqlType)]
#[diesel(postgres_type = "foo")]
struct Type6;

fn main() {}
