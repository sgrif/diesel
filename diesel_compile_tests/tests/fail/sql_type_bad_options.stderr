error: `postgres_type` is either unexpected or has bad format
 --> $DIR/sql_type_bad_options.rs:6:10
  |
6 | #[diesel(postgres_type)]
  |          ^^^^^^^^^^^^^

warning: unexpected `oid` when `type_name` is present
  --> $DIR/sql_type_bad_options.rs:10:49
   |
10 | #[diesel(postgres_type(type_name = "foo", oid = 2, array_oid = 3))]
   |                                                 ^

warning: unexpected `array_oid` when `type_name` is present
  --> $DIR/sql_type_bad_options.rs:10:64
   |
10 | #[diesel(postgres_type(type_name = "foo", oid = 2, array_oid = 3))]
   |                                                                ^

error: expected `oid` and `array_oid` or `type_name`
  --> $DIR/sql_type_bad_options.rs:13:10
   |
13 | #[derive(SqlType)]
   |          ^^^^^^^
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected integer literal
  --> $DIR/sql_type_bad_options.rs:18:45
   |
18 | #[diesel(postgres_type(oid = 1, array_oid = "1"))]
   |                                             ^^^

error: `ary_oid` is either unexpected or has bad format
  --> $DIR/sql_type_bad_options.rs:22:33
   |
22 | #[diesel(postgres_type(oid = 1, ary_oid = "1"))]
   |                                 ^^^^^^^

error: `postgres_type` is either unexpected or has bad format
  --> $DIR/sql_type_bad_options.rs:26:10
   |
26 | #[diesel(postgres_type = "foo")]
   |          ^^^^^^^^^^^^^
